storage/mysql/storage_test.go

```diff
@@ -889,6 +889,145 @@ func TestMapRootUpdate(t *testing.T) {
 	}
 }
 
+func TestMapSetGetRoundTrip(t *testing.T) {
+	cleanTestDB()
+
+	mapID := createMapID("TestMapSetGetRoundTrip")
+	db := prepareTestMapDB(mapID, t)
+	defer db.Close()
+	s := prepareTestMapStorage(mapID, t)
+
+	keyHash := []byte("A Key Hash")
+	value := trillian.MapLeaf{
+		LeafHash:  []byte("A Hash"),
+		LeafValue: []byte("A Value"),
+		ExtraData: []byte("Some Extra Data"),
+	}
+
+	readRev := int64(1)
+
+	{
+		tx := beginMapTx(s, t)
+
+		if err := tx.Set(keyHash, value); err != nil {
+			t.Fatalf("Failed to set %v to %v: %v", keyHash, value, err)
+		}
+		tx.Commit()
```

Comment:
Add checks for these new ones, but there are a *lot* of unchecked Commit() called in here, I've dropped a `TODO` in there to go back and sort those out too.